torch/linalg/__init__.py

```diff
@@ -140,3 +140,49 @@
     >>> LA.norm(m[0, :, :]), LA.norm(m[1, :, :])
     (tensor(3.7417), tensor(11.2250))
 """)
+
+tensorsolve = _add_docstr(_linalg.linalg_tensorsolve, r"""
+linalg.tensorsolve(input, other, dims=None, *, out=None) -> Tensor
+
+Computes a tensor ``x`` such that ``tensordot(input, x, dims=x.ndim) = other``.
+The resulting tensor ``x`` has the shape equal to ``input[other.ndim:]``.
+Before computing ``x``, dimensions of :attr:`input` can be moved using :attr:`dims` to match the shape requirements.
```

Comment:
Since the shape requirements haven't been introduced yet and "moved" is not well-defined, I suggest removing this sentence and leaving the documentation of dims to the more detailed note below.